File: spec/code_samples/PHP/v2-preview@reporting@messages@async@reports/get.php
<?php
$apiKey = 'YOUR_API_KEY';
$apiSecret = 'YOUR_API_SECRET';
$apiHost = 'YOUR_API_HOST'; // Set YOUR_API_HOST to the regional host from the servers section in the docs
$pageSize = 0;
$pageToken = "YOUR_PAGE_TOKEN";
$reportName = "YOUR_REPORT_NAME";
$status = "YOUR_STATUS";
$startDate = "YOUR_START_DATE";
$endDate = "YOUR_END_DATE";
$sortDirection = "YOUR_SORT_DIRECTION";

// HMAC authentication is also supported instead of Basic
$url = $apiHost . '/v2-preview/reporting/messages/async/reports?page_size=' . $pageSize . '&page_token=' . $pageToken . '&report_name=' . $reportName . '&status=' . $status . '&start_date=' . $startDate . '&end_date=' . $endDate . '&sort_direction=' . $sortDirection;
$headers = [
    'Authorization: Basic ' . base64_encode($apiKey . ':' . $apiSecret),
    'Accept: application/json',
];

$ch = curl_init($url);
curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'GET');
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

$response = curl_exec($ch);
$status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
curl_close($ch);

echo $status, PHP_EOL;
echo $response, PHP_EOL;

File: spec/code_samples/Ruby/v1@iam@signature_keys/get.rb
require 'net/http'
require 'json'
require 'base64'
require 'uri'

api_key = 'YOUR_API_KEY'
api_secret = 'YOUR_API_SECRET'
api_host = 'YOUR_API_HOST' # Set YOUR_API_HOST to the regional host from the servers section in the docs
page = "YOUR_PAGE"
page_size = "YOUR_PAGE_SIZE"

# HMAC authentication is also supported instead of Basic
uri = URI.parse("#{api_host}/v1/iam/signature_keys?page=#{page}&page_size=#{page_size}")
http = Net::HTTP.new(uri.host, uri.port)
http.use_ssl = uri.scheme == 'https'

request = Net::HTTP::Get.new(uri)
request['Authorization'] = 'Basic ' + Base64.strict_encode64("#{api_key}:#{api_secret}")
request['Accept'] = 'application/json'

response = http.request(request)
puts response.code
puts response.body
